rsocket/statemachine/RequestResponseResponder.cpp

```diff
@@ -37,7 +45,11 @@ void RequestResponseResponder::onSuccess(Payload response) noexcept {
 }
 
 void RequestResponseResponder::onError(folly::exception_wrapper ex) noexcept {
-  DCHECK(producingSubscription_);
+#ifdef DEBUG
+  DCHECK(gotOnSubscribe_.load()) << "didnt call onSubscribe";
+  DCHECK(!gotTerminating_.exchange(true)) << "Already called onSuccess/onError";
+#endif
+
```

Comment:
nit: do we need the same if check in here also?

Reply:
The next line unconditionally clears producingSubscriber

Reply:
But, what about the applicationError and closeStream calls? Do they try to send anything to the client side?